docs/section22/Rule22-9.md

```diff
@@ -0,0 +1,65 @@
+
+# CHW&CW - Rule 22-9  
+
+**Schema Version:** 0.0.10  
+**Mandatory Rule:** True  
+**Rule ID:** 22-9  
+**Rule Description:** For Baseline chilled water system with cooling capacity of 300 tons or more, the secondary loop shall be modeled with a minimum flow of 25% of the design flow rate.  
+**Rule Assertion:** B-RMR = expected value  
+**Appendix G Section:** Section 22 CHW&CW Loop  
+**90.1 Section Reference:** Section G3.1.3.10 Chilled-water pumps (System 7, 8, 11, 12 and 13)  
+**Data Lookup:** None  
+**Evaluation Context:** Building  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-7, 8, 11, 12, 13, 7b, 8b, 11b, 12b, 13b.
+2. B-RMR is modeled with primary-secondary configuration.
+3. B-RMR chilled water system cooling capacity is 300 tons or more.
+
+**Function Calls:**  
+
+1. get_baseline_system_types()
+2. get_primary_secondary_loops()
+3. get_component_by_id()
+
+**Applicability Checks:**  
+
+- **Check 1:** Get B-RMR system types: `baseline_hvac_system_dict = get_baseline_system_types(B-RMR)`
+
+  - Check if B-RMR is modeled with at least one air-side system that is Type-7, 8, 11.1, 11.2, 12, 13, 7b, 8b, 11b, 12b, 13b, i.e. with air-side system served by chiller(s), continue to the next applicability check: `if any(sys_type in baseline_hvac_system_dict.keys() for sys_type in ["SYS-7", "SYS-8", "SYS-11.1", "SYS-11.2", "SYS-12", "SYS-13", "SYS-7B", "SYS-8B", "SYS-11B", "SYS-12B", "SYS-13B"]): NEXT_APPLICABILITY_CHECK`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+- **Check 2:** Get primary and secondary loops for B-RMR: `primary_secondary_loop_dictionary = get_primary_secondary_loops(B_RMR, "COOLING")`
+
+  - Check if B-RMR is modeled with primary-secondary configuration, continue to rule logic: `if LEN(primary_secondary_loop_dictionary) != 0: CHECK_RULE_LOGIC`
+
+  - Else, rule is not applicable to B-RMR: `else: RULE_NOT_APPLICABLE`
+
+## Rule Logic:  
+
+- For each pump in B_RMR: `for pump_b in B_RMR.RulesetModelInstance.pumps:`
+
+  - Save loop served by pump and pump to a dictionary: `loop_pump_dictionary[pump_b.loop_or_piping].append(pump)`
+
+- For each chiller in B_RMR: `for chiller_b in B_RMR.RulesetModelInstance.chillers:`
+
```

Comment:
Project may have process chillers. So perhaps instead only consider chillers that have chiller.cooling_loop equal to the primary loop ID returned by get_primary_secondary_loops(B_RMR, "COOLING"). That is also not entirely full proof because process chiller may also have primary/secondary arrangement, but I don't think it's common so this method is less likely to misfire I think.

Reply:
as discussed during the RDS call on 2022.04.12 - the 'get primary secondary loops function' should be updated to exclude chilled water loops with process loads.

Reply:
get_primary_secondary_loops() function is now updated. 22-9 updated.